test/telemetry_metrics_statsd_test.exs

```diff
@@ -317,6 +317,7 @@ defmodule TelemetryMetricsStatsdTest do
       udp = TelemetryMetricsStatsd.get_udp(reporter)
```

Comment:
Can you explicitly start with the `pool_size: 1` in this test? Otherwise if we increase the default in the future we'll get a different socket in the assertion simply because there are many other sockets, regardless of the error handling logic.

Reply:
good point. Added